fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -685,7 +656,59 @@ private PlanNode createSelectPlan(SelectStmt selectStmt, Analyzer analyzer, long
         }
 
         // add aggregation, if required
-        if (aggInfo != null) root = createAggregationPlan(selectStmt, analyzer, root);
+        if (aggInfo != null) {
+            // introduce repeatNode for group by extension
+            GroupByClause groupByClause = selectStmt.getGroupByClause();
+            if (groupByClause != null && groupByClause.isGroupByExtension()) {
```

Comment:
I think that this kind of implementation don't work in some case.
If RepeatNode don't have itself's Tuple, you can set the group by item to null. 
For example query like `group by ROLLUP(a, ifnull(b, "a"))`, you only set b is null, but func(b) can be not null. which would make a wrong result